src/tool/hpcrun/gpu/opencl/opencl-api.c

```diff
@@ -1207,6 +1271,74 @@ hpcrun_clCreateBuffer
 }
 
 
+cl_int
+hpcrun_clWaitForEvents
+(
+	cl_uint num_events,
+	const cl_event* event_list
+)
+{
+	ETMSG(OPENCL, "clWaitForEvents called");
+	// on the assumption that clWaitForEvents is synchonous, we have sandwiched it with calls to sync_prologue and sync_epilogue
+	// clWaitForEvents can wait on multiple events(probably from different queues).
+	// We need a more complicated approach of finding the queues on which the CPU will wait
+	// For now we pass the 1st queue
+	void *queue_ptr;
+	cl_command_queue queue;
+
+	if(is_opencl_blame_shifting_enabled()) {
+		size_t queue_size;
+		clGetEventInfo(*event_list, CL_EVENT_COMMAND_QUEUE, 0, NULL, &queue_size);
+		queue_ptr = hpcrun_malloc(sizeof(queue_size));
```

Comment:
Where this `queue_ptr` is recycled?

Reply:
the queue_size is variable. A simple alloc and free queue won't work. how is dynamic-sized hpcrun_malloc freed in hpcrun?

Reply:
If it is in an application callback, just use malloc.
If only the current scope uses this variable, just declare it as a stack variable.